functions/src/common.ts

```diff
@@ -5,7 +5,7 @@ import { createHash } from "node:crypto";
 const FIRESTORE_WRITE_BATCH_SIZE = 450;
 
 export function safeString(unsafe: string):string {
-  return unsafe.replace(/\//gi, "_");
+  return unsafe.replace(/\//gi, '_').replace(/[^A-Za-z0-9- ]/g, '').trim();
```

Comment:
first replacement is obsolete and can be confusing

Reply:
original purpose of safeString is to generate safe document id for Firestore and maybe we might keep it as a separate function.
https://stackoverflow.com/questions/50383901/firestore-supported-characters-for-custom-ids

Reply:
maybe have to functions?

safeDocumentID - to guarantee document ID will be accepted: only Firestore safe symbols (looser restriction)

safeIndexingTarget - only alphanumeric + '-' + ' ' (tighter restriction)